python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -743,6 +761,101 @@ async def _compute_updated_since_time_subset(
         else:
             return self.get_full_subset(key=key)
 
+    @cached_method
+    async def _compute_latest_record(
+        self,
+        key: AssetKey,
+        of_type: Literal["observation", "materialization"],
```

Comment:
nit; feels like something we branch behavior on a bunch; should probably have some centralized way to distinguish easily